packages/ui/src/components/va-data-table/hooks/useTableScroll.ts

```diff
@@ -0,0 +1,63 @@
+import { getCurrentInstance, computed, ExtractPropTypes } from 'vue'
+
+import { useIntersectionObserver, useElementRef } from '../../../composables'
+
+export const useTableScrollProps = {
+  scrollTopMargin: { type: Number, default: 0 },
+  scrollBottomMargin: { type: Number, default: 0 },
+}
+
+export const useTableScrollEmits = ['scroll:top', 'scroll:bottom']
+
+export default function useTableScroll (
+  props: ExtractPropTypes<typeof useTableScrollProps>,
+  emit: (event: 'scroll:bottom' | 'scroll:top', ...args: any[]) => void,
+) {
+  // NOTE: unfortunately, this is not reactive
+  // TODO: replace 'vNodeProps' with '$listeners' when it's available in vue 3
+  const vNodeProps = getCurrentInstance()?.vnode.props
+  const isTopTriggerListener = vNodeProps?.['onScroll:top'] !== undefined
+  const isBottomTriggerListener = vNodeProps?.['onScroll:bottom'] !== undefined
+
+  const scrollContainer = useElementRef()
+  const topTrigger = useElementRef()
+  const bottomTrigger = useElementRef()
+
+  const isObservable = computed(() => !!scrollContainer.value)
+
+  const intersectionHandler = (entries: IntersectionObserverEntry[]) => {
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        entry.target === topTrigger.value
+          ? emit('scroll:top')
+          : emit('scroll:bottom')
+      }
+    })
+  }
+
+  const targets = computed<HTMLElement[]>(() => {
+    const list: HTMLElement[] = []
+
+    if (isObservable.value) {
```

Comment:
Do we need this check? This will be bound to tor and bottom triggers which is also DOM elements.